wordpress-mobile/WordPress-Android
# Send all translations to gutenberg-mobile

Requires https://github.com/wordpress-mobile/gutenberg-mobile/pull/536

This PR adds a new property our react app will receive which will contain all the translations available for WPAndroid. The idea is to have translations in gutenberg-mobile available for the beta.

### Testing Instructions

- Checkout this branch locally (`git fetch -a && git checkout add/i18n-cache`) and make sure to update the submodules after that `git submodule update --init --recursive`
- Remove this condition to test sending all the WPAndroid strings to gutenberg-mobile: https://github.com/wordpress-mobile/WordPress-Android/pull/9249/files#diff-02f567e707c1df878dfea548cfec3da7R151
- In a terminal run `cd libs/gutenberg-mobile && yarn install && yarn start:reset`
- Connect an android device or start a virtual device, then open another terminal, go to the WordPress-Android project root folder and run `./gradlew installWasabiDebug` then `adb shell am start -n "org.wordpress.android.beta/org.wordpress.android.ui.WPLaunchActivity" -a android.intent.action.MAIN -c android.intent.category.LAUNCHER` (or boot the app with Android studio)
- Make sure your app is in english
- Open http://localhost:8081/debugger-ui/ on your computer and open the console in the devtools
- Open a page or a post in the app
- Look at the logs and check that the locale is `en`, and that `translations` is empty
- Go to the user profile settings and switch to a different language (repeat the operation several times and test for languages with a regional code such as `zh-cn` or `pt-br`)
- Open a page or a post
- Look at the logs and check that the locale is correct (should be one of those https://github.com/wordpress-mobile/gutenberg-mobile/pull/536/files#diff-db716afdbf5160927b4465dba3e62535R8). For the locales for which we have a translation file in `libs/gutenberg-mobile/i18n-cache/data/` you should see the translated strings.
